doc/contributor/launchers.md

```diff
@@ -0,0 +1,38 @@
+# Executables in `bin` 
+
+## Requirements
+
+1.  [x] When TruffleRuby tar (GraalVM) is downloaded and unpacked, all files in `bin` and `language/ruby/bin` has to 
+    always run the TruffleRuby from the tar. 
+    -   Therefore `PATH` cannot be involved (`/usr/bin/env` cannot be used in shebang).
+    -   There cannot be absolute paths in `bin` executables.
+    -   All scripts in `bin` have to always resolve to `truffleruby` in the same dir.
+2.  [x] Newly installed gem executables has to always run the `truffleruby` from the same `bin`. 
+    -   By default RubyGems generate executables with absolute path.
+3.  [x] Behave as proper Ruby implementation when the `bin` is put on `PATH`.
+4.  [ ] RubyMine requires `gem` in `bin` to be Ruby (otherwise gems are not listed).
+    -   `gem` file in `bin` has to be Ruby or the `-x` option used by Rubymine has to be extended to understand our bash files in `bin`
+5.  [ ] `ruby -S irb` when executed in a `bin` directory does not work unless the `irb` file is Ruby file as well.
```

Comment:
The high level logic is in https://github.com/ruby/ruby/blob/a7db63208833d69ba02ef5ad21d32017f148bc46/ruby.c#L1502-L1514, which seems to indeed look up `$PATH` and otherwise trying to just execute the passed name.